### PR TITLE
Form: Required text of template editors overlaps the helpText property (T1107088)

### PR DESCRIPTION
**Pull request by EugeniyKiyashko**

### CODE REVIEW

File: js/ui/form/components/field_item.js [hunk 1]
```diff
@@ -207,3 +193,38 @@ export function renderFieldItem({
...
     return { $fieldEditorContainer, $rootElement, widgetInstance };
 }
+
+function getValidationTarget($fieldEditorContainer) {
+    const $editor = $fieldEditorContainer.children().first();
+    return $editor.hasClass(TEMPLATE_WRAPPER_CLASS) ? $editor.children().first() : $editor;
+}
+
+function tryGetValidationTargetInstance($validationTarget) {
+    return $validationTarget && $validationTarget.data(VALIDATION_TARGET_CLASS) || $validationTarget.parent().data(VALIDATION_TARGET_CLASS);
```
Comment (ksercs) [inline js/ui/form/components/field_item.js]:
```suggestion
    return $validationTarget?.data(VALIDATION_TARGET_CLASS) || $validationTarget?.parent?.()?.data(VALIDATION_TARGET_CLASS);
```

File: js/ui/form/components/field_item.js [hunk 2]
```diff
@@ -207,3 +193,38 @@ export function renderFieldItem({
...
+}
+
+function toggleWrapperInvalidClassInMaterial(validationTargetInstance) {
```
Comment (ksercs) [inline js/ui/form/components/field_item.js]:
```suggestion
function subscribeWrapperInvalidClassToggle(validationTargetInstance) {
```

File: js/ui/form/components/field_item.js [hunk 3]
```diff
@@ -207,3 +193,38 @@ export function renderFieldItem({
...
+                    isValid === false && (component._isFocused() || validationMessageMode === 'always')
+                );
+        };
+
+        validationTargetInstance.on('optionChanged', (e) => {
+            if(e.name !== 'isValid') return;
+            toggleInvalidClass(e);
+        });
+
+        validationTargetInstance
```
Comment (ksercs) [inline js/ui/form/components/field_item.js:225]: Where all this subscribed are removed?

File: js/ui/form/components/field_item.js [hunk 4]
```diff
@@ -162,27 +168,7 @@ export function renderFieldItem({
...
-                if(e.name !== 'isValid') return;
-                toggleInvalidClass(e);
-            });
-
-            validationTargetInstance
-                .on('focusIn', toggleInvalidClass)
-                .on('focusOut', toggleInvalidClass)
-                .on('enterKey', toggleInvalidClass);
-        }
+        toggleWrapperInvalidClassInMaterial(validationTargetInstance);
```
Comment (ksercs) [inline js/ui/form/components/field_item.js]: This row will cause double-subscribe if template is sync, won't it?

Comment (EugeniyKiyashko) [reply js/ui/form/components/field_item.js]: I reverted some changes after your review and the situation should be more predictable

File: js/ui/form/components/field_item.js (hunk 3 above)
Comment (EugeniyKiyashko) [reply js/ui/form/components/field_item.js:225]: Yes, I believe that is it

**Review by ksercs — APPROVED** [review]

**Review by marker-dao — APPROVED** [review]

**Pull request merged**